pkg/yurthub/cachemanager/cache_manager.go

```diff
@@ -301,7 +285,6 @@ func (cm *cacheManager) saveWatchObject(ctx context.Context, info *apirequest.Re
 			switch watchType {
 			case watch.Added, watch.Modified:
 				accessor.SetAPIVersion(obj, apiVersion)
-				accessor.SetKind(obj, kind)
```

Comment:
because after adding WithVersionCodecFactory, the GVK information is already in the object during decoding, actually the `SetAPIVersion`  don't need too.